### PR TITLE
Fix selection

### LINKED ISSUE #236: Issue with selecting arc/ellipse
--> Pick arc/ellipse from UI, OK
--> create the arc/ellipse ,OK
--> Click and drag cursor to cover the freshly created arc/ellipse in selection rectangle ,OK
!!! Nothing Happens

### PR DESCRIPTION
**Pull request by rupakbajgain**
-Old selection doesn't give 'What you see is what you get'.
Does this fix #236 , it worked for me?

### CODE REVIEW

**Comment by rupakbajgain** [comment]
@feragon , i removed your merged contents because i had problem with push, then i pulled new commits from master branch. I think you did this because old code cannot be merged.

**Comment by rupakbajgain** [comment]
This fixes #240 .

**Comment by rupakbajgain** [comment]
This fixes #240 .

**Comment by rupakbajgain** [comment]
does it fix #235 too, it worked for me or was previously fixed.

**Comment by feragon** [comment]
Ok for #235 and #240

**Comment by rupakbajgain** [comment]
#108 is fixed too, we need to preserve block information also.

File: lcviewernoqt/documentcanvas.cpp [hunk 1]
```diff
@@ -523,29 +523,31 @@ void DocumentCanvas::makeSelection(double x, double y, double w, double h, bool
...
     }
     else {
         entitiesInSelection = entityContainer().entitiesWithinAndCrossingArea(*_selectedArea);
     }
-    _newSelection.clear();
     entitiesInSelection.each< const lc::entity::CADEntity >([&](lc::entity::CADEntity_CSPtr entity) {
         auto di = _entityDrawItem[entity->id()];
-        auto iter = std::find(_selectedDrawables.begin(), _selectedDrawables.end(), di);
-        _newSelection.push_back(di);
-        di->selected(!entity || iter == _selectedDrawables.end());
```
Comment (rupakbajgain) [inline lcviewernoqt/documentcanvas.cpp:548]: Why is !entity here?

Comment (rupakbajgain) [reply lcviewernoqt/documentcanvas.cpp:548]: it is in old code, i removed in new. should i keep it.

**Comment by feragon** [comment]
Ok for #108 too.
I keep #236 open as we discovered a new bug. Thanks for your work :+1:

**Pull request merged**